<snippet><description>throw 语句</description><tabTrigger>gra-error throw</tabTrigger><scope>source.js</scope><content><![CDATA[
/**
 * throw 语句的作用是手动中断程序执行，抛出一个错误。
 */
x = -1
if (x <= 0) {
  throw new Error('x 必须为正数');
}
// Error: x 必须为正数 ，整个程序就会在这里中断执行

/**
 * throw也可以抛出自定义错误。
 */
function UserError(message) {
  this.message = message || '默认信息';
  this.name = 'UserError';
}
UserError.prototype = new Error();
UserError.prototype.constructor = UserError;

throw new UserError('出错了！');
// UserError: 出错了！

/**
 * 实际上，throw可以抛出任何类型的值。也就是说，它的参数可以是任何值。
 */
throw 'Error！'; // Uncaught Error！ 抛出一个字符串
throw 42; // Uncaught 42 抛出一个数值
throw true; // Uncaught true 抛出一个布尔值

throw {
  toString: function () {
    return 'Error!';
  }
}; // Uncaught {toString: ƒ} 抛出一个对象
]]></content></snippet>